### ISSUE TITLE
Issue with Kubernetes Ingress Object Field: servicePort

### ISSUE THREAD

**Issue by gallowaystorm** [hashicorp/terraform-provider-kubernetes#1547]
### Terraform Version, Provider Version and Kubernetes Version

```
Terraform version: 1.1.0
Kubernetes provider version: provider registry.terraform.io/hashicorp/kubernetes v2.7.1
Kubernetes version: 1.21
```

### Affected Resource(s)

- kubernetes_manifest

### Terraform Configuration Files
```terraform
# Get certificate to use for ALB
data "aws_acm_certificate" "enivronment_default_issued" {
  domain = "*.${var.environment}.${var.team_name}.one.singledigits.com"
}


resource "kubernetes_manifest" "service_ingress_object" {
  manifest = {
    "apiVersion" = "networking.k8s.io/v1beta1"
    "kind"       = "Ingress"
    "metadata" = {
      "name"      = "${var.team_name}-app-mesh-ingress"
      "namespace" = "${var.kubernetes_namespace}"
      "annotations" = {
        "alb.ingress.kubernetes.io/actions.ssl-redirect" = "{\"Type\": \"redirect\", \"RedirectConfig\": { \"Protocol\": \"HTTPS\", \"Port\": \"443\", \"StatusCode\": \"HTTP_301\"}}"
        "alb.ingress.kubernetes.io/listen-ports"         = "[{\"HTTP\": 80}, {\"HTTPS\":443}]"
        "alb.ingress.kubernetes.io/certificate-arn"      = "${data.aws_acm_certificate.enivronment_default_issued.arn}"
        "alb.ingress.kubernetes.io/scheme"               = "internal"
        "alb.ingress.kubernetes.io/target-type"          = "instance"
        "kubernetes.io/ingress.class"                    = "alb"
      }
    }
    "spec" = {
      "rules" = [{
        "host" = "${var.service_name}.${var.dns_zone_name}"
        "http" = {
          "paths" = [
            {
              "path" = "/*"
              "backend" = {
                "serviceName" = "ssl-redirect"
                "servicePort" = "use-annotation"
              }
            },
            {
              "path" = "/*"
              "backend" = {
                "serviceName" = "${var.service_name}-svc"
                "servicePort" = "${var.service_port}"
              }
            },
          ]
        }
      }]
    }
  }
}
```

### Debug Output

sd
### Panic Output

```bash
panic: lists must only contain one type of element, saw tftypes.Object["backend":tftypes.Object["resource":tftypes.Object["apiGroup":tftypes.String, "kind":tftypes.String, "name":tftypes.String], "serviceName":tftypes.String, "servicePort":tftypes.String], "path":tftypes.String, "pathType":tftypes.String] and tftypes.Object["backend":tftypes.Object["resource":tftypes.Object["apiGroup":tftypes.String, "kind":tftypes.String, "name":tftypes.String], "serviceName":tftypes.String, "servicePort":tftypes.Number], "path":tftypes.String, "pathType":tftypes.String]
```

### Steps to Reproduce

1. `terraform apply -var-file=./application.tfvars` 
2. select "yes" 

### Expected Behavior
What should have happened?

The ingress object should have deployed an ingress object with a servicePort of 443 and an ssl redirect on the AWS load balancer for all other port listeners.

### Actual Behavior
What actually happened?

```bash
│ Error: Plugin did not respond
│ 
│   with module.services.module.portal.module.appmesh.kubernetes_manifest.service_ingress_object,
│   on .terraform/modules/services.portal.appmesh/kubernetes_manifest.tf line 104, in resource "kubernetes_manifest" "service_ingress_object":
│  104: resource "kubernetes_manifest" "service_ingress_object" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).PlanResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-kubernetes_v2.7.1_x5 plugin:

panic: lists must only contain one type of element, saw tftypes.Object["backend":tftypes.Object["resource":tftypes.Object["apiGroup":tftypes.String, "kind":tftypes.String, "name":tftypes.String], "serviceName":tftypes.String, "servicePort":tftypes.String], "path":tftypes.String, "pathType":tftypes.String] and tftypes.Object["backend":tftypes.Object["resource":tftypes.Object["apiGroup":tftypes.String, "kind":tftypes.String, "name":tftypes.String], "serviceName":tftypes.String, "servicePort":tftypes.Number], "path":tftypes.String, "pathType":tftypes.String]

goroutine 567 [running]:
github.com/hashicorp/terraform-plugin-go/tftypes.NewValue(...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tftypes/value.go:277
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.morphTupleIntoType(0x26d6380, 0xc002ffb470, 0x1f48820, 0xc001a6e570, 0x26d6260, 0xc0010a3110, 0xc005e4d8d8, 0x1100000000000000, 0x203000, 0x203000, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/morph.go:201 +0x1b85
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.ValueToType(0x26d6380, 0xc002ffb470, 0x1f48820, 0xc001a6e570, 0x26d6260, 0xc0010a3110, 0xc005e4d8d8, 0x8, 0x26d63c8, 0xc0013583f0, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/morph.go:30 +0xa09
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.morphObjectToType(0x26d62f0, 0xc006c862a0, 0x2058260, 0xc002ff6990, 0x26d62f0, 0xc0010a3140, 0xc005e4d8c0, 0x203000, 0xc002ff67b0, 0x1f98920, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/morph.go:314 +0x425
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.ValueToType(0x26d62f0, 0xc006c862a0, 0x2058260, 0xc002ff6990, 0x26d62f0, 0xc0010a3140, 0xc005e4d8c0, 0x6, 0xc002ff67b0, 0x1f98920, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/morph.go:36 +0x709
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.morphObjectToType(0x26d62f0, 0xc006c872f0, 0x2058260, 0xc002ff6690, 0x26d62f0, 0xc0010a31a0, 0xc005e4d848, 0x11000000026d63c8, 0xc001301b90, 0x8, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/morph.go:314 +0x425
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.ValueToType(0x26d62f0, 0xc006c872f0, 0x2058260, 0xc002ff6690, 0x26d62f0, 0xc0010a31a0, 0xc005e4d848, 0x4, 0xc0010629f0, 0xc005e4d7d0, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/morph.go:36 +0x709
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.morphTupleIntoType(0x26d6380, 0xc006c8a570, 0x1f48820, 0xc001a6eb10, 0x26d6260, 0xc0010a31d0, 0xc005e4d800, 0x10, 0x100, 0x80, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/morph.go:195 +0xddb
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.ValueToType(0x26d6380, 0xc006c8a570, 0x1f48820, 0xc001a6eb10, 0x26d6260, 0xc0010a31d0, 0xc005e4d800, 0x2, 0x0, 0x0, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/morph.go:30 +0xa09
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.morphObjectToType(0x26d62f0, 0xc006c8b8c0, 0x2058260, 0xc002ff6300, 0x26d62f0, 0xc0006b8000, 0xc005e4d7e8, 0x26d62f0, 0xc0010629f0, 0x2058260, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/morph.go:314 +0x425
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.ValueToType(0x26d62f0, 0xc006c8b8c0, 0x2058260, 0xc002ff6300, 0x26d62f0, 0xc0006b8000, 0xc005e4d7e8, 0x1, 0xc0010629f0, 0x2058260, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/morph.go:36 +0x709
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.morphObjectToType(0x26d62f0, 0xc006c955c0, 0x2058260, 0xc002fee6c0, 0x26d62f0, 0xc0016f2a50, 0xc005e4d1d0, 0x1100000000000000, 0x0, 0xff4004c, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/morph.go:314 +0x425
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.ValueToType(0x26d62f0, 0xc006c955c0, 0x2058260, 0xc002fee6c0, 0x26d62f0, 0xc0016f2a50, 0xc005e4d1d0, 0x2, 0x7, 0x0, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/morph.go:36 +0x709
github.com/hashicorp/terraform-provider-kubernetes/manifest/provider.(*RawProviderServer).PlanResourceChange(0xc00057e280, 0x26c81e8, 0xc002374f00, 0xc001212500, 0xc00044a978, 0x1, 0x30)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/provider/plan.go:315 +0x1cbb
github.com/hashicorp/terraform-plugin-mux.SchemaServer.PlanResourceChange(0xc00057c930, 0xc00057c960, 0xc0018666c0, 0x2, 0x2, 0xc00057e180, 0x1, 0x26c81e8, 0xc002374f00, 0xc001212500, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-mux/schema_server.go:276 +0x98
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).PlanResourceChange(0xc0018666e0, 0x26c8290, 0xc002374f00, 0xc001e1bd50, 0xc0018666e0, 0xc00270a150, 0xc00188dba0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:315 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler(0x2236720, 0xc0018666e0, 0x26c8290, 0xc00270a150, 0xc00236a540, 0x0, 0x26c8290, 0xc00270a150, 0xc006dde000, 0x2525)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:362 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00038e8c0, 0x26e0078, 0xc0012af080, 0xc0022ef7a0, 0xc00057c9f0, 0x3556c48, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:1217 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00038e8c0, 0x26e0078, 0xc0012af080, 0xc0022ef7a0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:1540 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00027c200, 0xc00038e8c0, 0x26e0078, 0xc0012af080, 0xc0022ef7a0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:878 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:876 +0x1fd

Error: The terraform-provider-kubernetes_v2.7.1_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2021-12-13T13:34:12.542-0600 [DEBUG] provider: plugin exited
2021-12-13T13:34:12.543-0600 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-12-13T13:34:12.543-0600 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-12-13T13:34:12.544-0600 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/time/0.7.2/linux_amd64/terraform-provider-time_v0.7.2_x5 pid=9035
2021-12-13T13:34:12.544-0600 [DEBUG] provider: plugin exited
2021-12-13T13:34:12.545-0600 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/3.69.0/linux_amd64/terraform-provider-aws_v3.69.0_x5 pid=9054
2021-12-13T13:34:12.545-0600 [DEBUG] provider: plugin exited
```

### Important Factoids

None

### References

Not that I have found
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jj-mino** [comment]
It also fails at previous kubernetes providers (2.6.1).
It seems that you cannot provide `servicePort` as a number, but it works if you just use `http` as the value

**Comment by artm** [comment]
this is a duplicate of https://github.com/hashicorp/terraform-provider-kubernetes/issues/1359

**Comment by alexsomesan** [comment]
This type of error was fixed in provider version 2.8.0 
https://github.com/hashicorp/terraform-provider-kubernetes/releases/tag/v2.8.0
